src/cmds/repo.rs

```diff
@@ -0,0 +1,73 @@
+use crate::cheat::SuggestionType;
+use crate::filesystem;
+use crate::fzf;
+use std::error::Error;
+use std::fs;
+use std::io::Write;
+use std::process::{Command, Stdio};
+
+pub fn add(uri: String) -> Result<(), Box<dyn Error>> {
+    let mut parts = uri.split('/');
+    let user = parts.next().unwrap();
+    let repo = parts.next().unwrap();
+    let cheat_path_str = filesystem::pathbuf_to_string(filesystem::cheat_pathbuf().unwrap());
+    let repo_folder_str = format!("{}-master", repo);
+    let tmp_path_str = format!("{}/tmp", cheat_path_str);
+
+    // TODO: stop shell'ing out
+    let cmd = format!(
+        r#"
+      url="https://github.com/{user}/{repo}/archive/master.tar.gz";
+      mkdir -p "{tmp_path}"; 
+      cd "{tmp_path}"; 
+      (wget -c "$url" -O - || curl -L "$url") | tar -xz || exit 42;
```

Comment:
Is this compatible with most systems? :thinking:

Reply:
Any reason not to use [`git clone --depth 1`](https://git-scm.com/docs/git-clone#Documentation/git-clone.txt---depthltdepthgt)?

`git` command will probably be available in all platforms, plus you don't need to extract the gz file later, yet you don't get all of the git history, just the latest commit.
In the future, it'll be easier to get the latest changes with `git pull --depth/--deepen` without having to download each repo all over again.

Reply:
Because I didn't want to add dependencies for a curl-like and tar-like operations. I may reconsider it though

Reply:
This way there is a silent dependency on the above commands, assuming they're present. Some systems, *and users*, like to do weird stuff :man_shrugging:

Reply:
Regarding git clone: fewer systems have git (compared to curl/wget) + the repo source code is temporary the way I implemented, so a git pull wouldn't hep